_test/general.test.php

```diff
@@ -58,4 +78,408 @@ public function test_plugin_conf() {
+        $ordernumber = $ordernumbers[0];
+
+        $attributes = $xml->xpath('/findologic/items/item/allAttributes/attributes/attribute');
+        $attributeKey = (string)$attributes[0]->key[0];
+        $attributeValue = (string)$attributes[0]->values[0]->value[0];
+
+        $expectedOrdernumber = $pageId;
+        $expectedAttributeKey = 'cat';
+        $expectedAttributeValue = 'test123_test123';
+
+        $this->assertEquals($expectedAttributeKey, $attributeKey, 'Expected attribute key in XML should match the category value when namespace height is two "cat".');
+        $this->assertEquals($expectedAttributeValue, $attributeValue, 'Expected attribute value in XML should match the namespace formatted in a FINDOLOGIC proper format when namespace height is two "test123_test123".');
+        $this->assertEquals($expectedOrdernumber, $ordernumber, 'Expected ordernumber in XML should match the pages namespace when namespace height is two "test123:test123".');
+    }
+
+    /**
+     * Test to ensure that ordernumber and attribute categories in the XML are valid when DokuWiki has a namespace height of one.
+     */
+    public function test_xml_elements_equal_to_dokuwiki_page_data_when_namespace_height_is_one() {
+        $pageId = 'test321';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(0, 20));
+
+        $ordernumbers = $xml->xpath('/findologic/items/item/allOrdernumbers/ordernumbers/ordernumber');
+        $ordernumber = $ordernumbers[0];
+
+        $attributes = $xml->xpath('/findologic/items/item/allAttributes/attributes/attribute');
+        $attributeKey = (string)$attributes[0]->key[0];
+        $attributeValue = (string)$attributes[0]->values[0]->value[0];
+
+        $expectedOrdernumber = $pageId;
+        $expectedAttributeKey = 'cat';
+        $expectedAttributeValue = 'test321';
+
+        $this->assertEquals($expectedAttributeKey, $attributeKey, 'Expected attribute key in XML should match the category value when namespace height is two "cat".');
+        $this->assertEquals($expectedAttributeValue, $attributeValue, 'Expected attribute value in XML should match the namespace formatted in a FINDOLOGIC proper format when namespace height is two "test123_test123".');
+        $this->assertEquals($expectedOrdernumber, $ordernumber, 'Expected ordernumber in XML should match the pages namespace when namespace height is two "test123:test123".');
+    }
+
+    //======================================================================
+    // EXPORT CALL TESTS
+    //======================================================================
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = 1, count = 1, total = 1;
+     * @expectedException InvalidArgumentException
```

Comment:
This works fine. However, there may be cases where it makes sense to manually catch the exception in the test so you can add an asssertion message for the failure case, eg.

```php
try {
  $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(1, 1));
  $this->fail('An exception should be thrown when ...`);
} catch (\InvalidArgumentException $e) {
}
```

Reply:
Oh yes! Thanks for this input, because the moment I was writing this test I was like "Hm.. how can I set an assertion message if this test fails..?". :)